diff --git a/auto/lib/pcre/conf b/auto/lib/pcre/conf
index 5e3960f..e32dfa5 100644
--- a/auto/lib/pcre/conf
+++ b/auto/lib/pcre/conf
@@ -78,9 +78,10 @@ if [ $PCRE != NONE ]; then
                 have=PCRE_STATIC . auto/have
             fi
 
-            CORE_DEPS="$CORE_DEPS $PCRE/pcre.h"
-            LINK_DEPS="$LINK_DEPS $PCRE/.libs/libpcre.a"
-            CORE_LIBS="$CORE_LIBS $PCRE/.libs/libpcre.a"
+            #CORE_DEPS="$CORE_DEPS $PCRE/pcre.h"
+            #LINK_DEPS="$LINK_DEPS $PCRE/.libs/libpcre.a"
+            #CORE_LIBS="$CORE_LIBS $PCRE/.libs/libpcre.a"
+            CORE_LIBS="$CORE_LIBS -L$PCRE_BIN -lpcre"
         ;;
 
     esac
diff --git a/auto/lib/zlib/conf b/auto/lib/zlib/conf
index 239592e..a5a49f6 100644
--- a/auto/lib/zlib/conf
+++ b/auto/lib/zlib/conf
@@ -33,9 +33,9 @@ if [ $ZLIB != NONE ]; then
 
         *)
             have=NGX_ZLIB . auto/have
-            LINK_DEPS="$LINK_DEPS $ZLIB/libz.a"
-            CORE_LIBS="$CORE_LIBS $ZLIB/libz.a"
-            #CORE_LIBS="$CORE_LIBS -L $ZLIB -lz"
+            #LINK_DEPS="$LINK_DEPS $ZLIB/libz.a"
+            #CORE_LIBS="$CORE_LIBS $ZLIB/libz.a"
+            CORE_LIBS="$CORE_LIBS -L$ZLIB_BIN -lz"
         ;;
 
     esac
